src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3765,6 +3791,10 @@ mono_emit_simd_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector")){
 		return emit_sri_vector (cfg, cmethod, fsig, args);
 	}
+
+	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector`1")){
+		return emit_vector64_vector128_t (cfg, cmethod, fsig, args);
+	}
 #endif // defined(TARGET_ARM64)
 
 	return emit_simd_intrinsics (class_ns, class_name, cfg, cmethod, fsig, args);
```

Comment:
As discussed, emit_simd_intrinsics () should be renamed arch_emit_simd_intrinsics () or something, and the code inside the ifdefs above should go into that function. Alternative, the code inside emit_simd_intrinsics () should go into this function.

Reply:
Yes, I will refactor once I fix all the errors. It will be included in this PR for sure.